cogs/servercontrol.py

```diff
@@ -0,0 +1,135 @@
+import asyncio
+import socket
+import discord
+from discord.ext.commands import Cog, command, has_any_role
+from utils import load_config, cog_error_handler
+from utils.discord import not_nobot_role
+import asyncssh
+
+global_config = load_config()
+config = global_config
+
+class server(Cog):
+
+    cog_command_error = cog_error_handler
+
+    #server command: start, stop, restart, update
+    @command(help=config.cogs.servercontrol.server.help)
+    @has_any_role(*config.cogs.servercontrol.server.role_names)
+    async def server(self, ctx, server, command):
+
+        servers = ["eu", "us", "eumvm", "usmvm"]
+        commands = ["start", "stop", "restart", "update", "ping"]
+
+        if command in commands:
+            if server in servers:
+                #eu
+                if server == "eu":
+                    await ctx.send(await server_connection("tf", "eu.tf2maps.net", 27015, command))
+                #us
+                elif server == "us":
+                    await ctx.send(await server_connection("tf", "us.tf2maps.net", 27015, command))
+
+                #eumvm
+                elif server == "eumvm":
+                    await ctx.send(await server_connection("mvm", "eu.tf2maps.net", 27016, command))
+
+                #usmvm
+                elif server == "usmvm":
+                    await ctx.send(await server_connection("mvm", "us.tf2maps.net", 27016, command))
+
+                else:
+                    await ctx.send("How did it reach this part?")
+
+            else:
+                await ctx.trigger_typing()
+                await ctx.send("Server not found.")
+
+        else:
+            await ctx.send("Invalid command argument.")
+
+#server connecting
+async def server_connection(user, hostname, port, command):
+
+    commands = ["start", "stop", "restart", "update", "ping"]
+
+    async with asyncssh.connect(
+        hostname,
+        port=22, 
+        username=user, 
+        password=config.sftp.master.password,
+        known_hosts=None
+    ) as conn:
+        if command in commands:
+            if command == "start":
+
+                await conn.run('./server start')
```

Comment:
we probably will want to do this another way; this seems ripe for errors

We would also want a way to poll for the statusd of the server instead of just hping it comes up in 8 seconds

Reply:
Going to need your insight on this one.